plinth/utils.py

```diff
@@ -52,6 +52,10 @@ def non_admin_view(func):
     return func
 
 
-def is_user_admin(user):
+def is_user_admin(request, cached=False):
     """Return whether user is an administrator."""
-    return user.groups.filter(name='admin').exists()
+    if cached and request.session['cache_user_is_admin']:
```

Comment:
This `if` condition will result in cache not being used and an SQL query being made whenever the user is not an administrator (most of the time).  You need to distinguish `None` value form a `False` value.  Use cache for `False` value but not for `None` value.